examples/gno.land/p/demo/uuid/uuid.gno

```diff
@@ -0,0 +1,132 @@
+// Package uuid provides utility functions for generating unique identifiers
+// (UUIDs) based on a custom implementation inspired by the Snowflake algorithm.
+// The UUIDs generated by this package ensure uniqueness and are suitable for
+// distributed systems. The package offers a mechanism to convert these IDs
+// to standard UUID format strings, making them compatible with systems
+// requiring UUIDs.
+// add import "gno.land/p/demo/entropy" to use the entropy package. For of the pseudo-random number generator.
+
+package uuid
+
+import (
+	"encoding/binary"
+	"std"
+	"strings"
+	"time"
+
+	"gno.land/p/demo/entropy"
+	"gno.land/p/demo/ufmt"
+)
+
+const (
+	TimestampLength uint8  = 41
+	MachineIDLength uint64 = 10
+	SequenceLength  uint64 = 12
+	MaxSequence     uint16 = 1<<SequenceLength - 1
+	MaxTimestamp    uint64 = 1<<TimestampLength - 1
+	MaxMachineID    uint64 = 1<<MachineIDLength - 1
+
+	machineIDMoveLength = SequenceLength
+	timestampMoveLength = MachineIDLength + SequenceLength
+)
+
+// UUID struct for generating unique identifiers.
+type UUID struct {
+	MachineID     uint64
+	sequence      uint16
+	StartTime     int64
+	LastTimestamp int64
+	entropy       *entropy.Instance
+}
+
+func NewUUID() *UUID {
+	u := &UUID{
+		StartTime: time.Date(2008, 11, 10, 23, 0, 0, 0, time.UTC).Unix() * 1000,
```

Comment:
If this is fixed, why not just use the constant unix time for this value?

Also, what is this value?

Reply:
Indeed you are right it would be better to provide the data directly in `Unix` which would avoid recalculating them at each function call.
The value used is the one used in `snowflake` because it is based on the fact that `go playground` starts with the temp of `2009-11-10 23:00:00 UTC`.
So I took the same as that of snowflake because I did not find the time.Now() of `gno playground` and I believe that in `gno` [time.Now()](https://docs.gno.land/reference/go-gno-compatibility/#tooling-gno-binary) is based on the return of the block time rather than the system time.
<img width="564" alt="SCR-20240820-pcie" src="https://github.com/user-attachments/assets/e5ebd8e1-f96d-4b66-b362-02e3d16f6065">


Do you have more information on this subject?

Reply:
>in gno [time.Now()](https://docs.gno.land/reference/go-gno-compatibility/#tooling-gno-binary) is based on the return of the block time rather than the system time.

Yes, it's the block execution context time, and same for all transactions within a block

Can you add a comment, and change this value to unix?

Reply:
> > in gno [time.Now()](https://docs.gno.land/reference/go-gno-compatibility/#tooling-gno-binary) is based on the return of the block time rather than the system time.
> 
> Yes, it's the block execution context time, and same for all transactions within a block
> 
> Can you add a comment, and change this value to unix?

Yep, I had already made this modification in this commit :) 
[353beb1](https://github.com/gnolang/gno/pull/2076/commits/353beb1f637e24bf513526e6d28976cd8a13cb41)